### PR TITLE
remove assert_allclose from torch.testing

### PR DESCRIPTION
**Pull request by pmeier**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #87974
* #87973
* #87972
* #87971
* #87970
* #87969

See #87969 or #86586 for the reasoning.

cc @albanD

### CODE REVIEW

File: caffe2/python/operator_test/_utils.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 """
-This file only exists since `torch.testing.assert_allclose` is deprecated, but used extensively throughout the tests in
+This file only exists since `torch.testing.assert_allclose` was removed, but used extensively throughout the tests in
```
Comment (mruberry) [inline caffe2/python/operator_test/_utils.py:2]: Nice doc update

**Review by mruberry — APPROVED** [review]
Nice to have the deprecations finalized!

**Comment by albanD** [comment]
IIRC pytest is swallowing warnings?
(this is one of the few reasons why I would not recommend using pytest locally).

**Comment by NicolasHug** [comment]
> (this is one of the few reasons why I would not recommend using pytest locally).

Out of curiosity what would you recommend instead?

**Comment by NicolasHug** [comment]
> IIRC pytest is swallowing warnings?

Yeah, this is due to:

https://github.com/pytorch/pytorch/blob/35be73df094f02dd26562cf665a6158e80bc4045/pytest.ini#L10

By default, pytest does show the warnings though. So unless users have manually silenced their pytest warnings (like pytorch), they should have properly seen the deprecation message

**Comment by pmeier** [comment]
From https://github.com/pytorch/pytorch/pull/87969#issue-1426801031:

> Deprecation happened on Feb 25, 2022 in c6f1bbc0ac33be0c8ad9956e3fc15e78ddb6cb95, which made it into the 1.12 release. Since it is now 245 days later and the next release will be 1.14, the removals later in the stack comply with the [BC policy](https://github.com/pytorch/pytorch/wiki/PyTorch's-Python-Frontend-Backward-and-Forward-Compatibility-Policy#minimizing-the-disruption-of-bc-breaking-changes).

The RFC that lead to this deprecation is in #61844.

**Comment by albanD** [comment]
> Out of curiosity what would you recommend instead?

tbh I use the direct `python test_foo.py` locally (with the additional -f -k or -v as needed).
It feels a lot more baremetal and no-magic-added which I prefer (but it is very personal!).

**Comment by xuzhao9** [comment]
This PR breaks downstream code such as https://github.com/kornia/kornia/blob/master/kornia/testing/__init__.py#L251

**Comment by albanD** [comment]
@xuzhao9 sorry about that. I'm a bit confused as the code there seem to imply that this case is only used for PT <1.9 ?

**Comment by xuzhao9** [comment]
@albanD The authors are aware of this change and is discussing a mitigation: https://github.com/kornia/kornia/issues/1981
The offending line also includes: https://github.com/kornia/kornia/blob/v0.6.8/kornia/testing/__init__.py#L235.

**Comment by mehtanirav** [comment]
Unfortunately, we need to revert this PR. We have many references to `torch.testing.assert_allclose` in internal tests that need to be updated before we can re-land this PR. 

I will follow up on the task to update all internal reference to `torch.testing.assert_allclose` and inform @pmeier when it safe to re-land the PR.

@pytorchbot revert -m "Internal breakages from method removal" -c ghfirst

**Comment by mehtanirav** [comment]
cc: @albanD on the revert.

**Pull request closed without merging**